transient/sshfs.py

```diff
@@ -115,127 +108,92 @@ def provision_system(timeout: int, ssh_config: ssh.SshConfig, is_slow: bool) ->
+        # Check stderr
+        stderr = ssh_proc.communicate()[1].decode("utf-8")
         if _should_provision(is_provisioned, stderr) is False:
             raise RuntimeError(f"SSHFS mount failed with: {stderr}")
 
-        # If so, try to provision it
-        success = provision_system(connect_timeout, ssh_config, is_slow)
-        if success is True:
-            # Try the sshfs again once we have provisioned the sytem
-            do_sshfs_mount(
-                connect_timeout=connect_timeout,
-                local_dir=local_dir,
-                remote_dir=remote_dir,
-                local_user=local_user,
-                local_password=local_password,
-                ssh_config=ssh_config,
-                is_provisioned=True,
-            )
-        else:
-            raise RuntimeError(
-                f"Unable to provision and SSHFS mount failed with: {stderr}"
-            )
-    except subprocess.TimeoutExpired:
-        # The timeout expired (as expected), but because we 'set -e', this means
-        # we must be in the state where we're hanging after the logout. So kill
-        # the connection from our end, the sshfs process will continue on the
-        # guest side.
-        conn.terminate()
+        success = provision_system(connect_timeout, ssh_config)
+        if not success:
+            raise RuntimeError(f"Unable to provision. SSHFS mount failed with: {stderr}")
+
+        # Unprovisioned systems do not touch stdout, so sftp_proc can be reused.
+        do_sshfs_mount(
+            connect_timeout=connect_timeout,
+            local_dir=local_dir,
+            remote_dir=remote_dir,
+            ssh_config=ssh_config,
+            sftp_proc=sftp_proc,
+            is_provisioned=True,
+        )
 
-        # There is a chance the sshfs process hung, so check for the sentinel text
-        raw_stdout, raw_stderr = conn.communicate()
-        stdout = raw_stdout.decode("utf-8")
-        stderr = raw_stderr.decode("utf-8")
-        if "TRANSIENT_SSHFS_DONE" not in stdout:
-            if is_slow:
-                # We timed out without getting to the 'echo' in the script, even with
-                # extra time. Just give up.
-                raise RuntimeError(f"SSHFS mount timed out: {stderr}")
-            else:
-                logging.warning(
-                    "sshfs mount did not complete. Trying with longer timeout"
-                )
-                # Try again with a longer timeout. The system may just be extremely
-                # slow (like qemu with 1 core and no kvm acceleration)
-                do_sshfs_mount(
-                    connect_timeout=connect_timeout,
-                    local_dir=local_dir,
-                    remote_dir=remote_dir,
-                    local_user=local_user,
-                    local_password=local_password,
-                    ssh_config=ssh_config,
-                    is_slow=True,
-                )
+    # We already waited for the timeout
```

Comment:
I'd like a little more description from this. Maybe "Sanity check that the ssh connection has not closed unexpectedly before we return"

Reply:
I agree. I have improved this.